test/executor/drop_test.cpp

```diff
@@ -219,5 +220,68 @@ TEST_F(DropTests, DroppingTrigger) {
   txn_manager.CommitTransaction(txn);
 }
 
+TEST_F(DropTests, DroppingIndexByName) {
+  auto catalog = catalog::Catalog::GetInstance();
+  catalog->Bootstrap();
+
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+
+  catalog->CreateDatabase(TEST_DB_NAME, txn);
+  auto db = catalog->GetDatabaseWithName(TEST_DB_NAME, txn);
+  // Insert a table first
+  auto id_column = catalog::Column(
+      type::TypeId::INTEGER, type::Type::GetTypeSize(type::TypeId::INTEGER),
+      "dept_id", true);
+  auto name_column =
+      catalog::Column(type::TypeId::VARCHAR, 32, "dept_name", false);
+
+  std::unique_ptr<catalog::Schema> table_schema(
+      new catalog::Schema({id_column, name_column}));
+  std::unique_ptr<catalog::Schema> table_schema2(
+      new catalog::Schema({id_column, name_column}));
+  txn_manager.CommitTransaction(txn);
+
+  txn = txn_manager.BeginTransaction();
+  catalog->CreateTable(TEST_DB_NAME, "department_table_01",
+                       std::move(table_schema), txn);
+  txn_manager.CommitTransaction(txn);
+
+  txn = txn_manager.BeginTransaction();
+  auto source_table = db->GetTableWithName("department_table_01");
+  oid_t col_id = source_table->GetSchema()->GetColumnID(
+              id_column.column_name);
+  std::vector<oid_t> source_col_ids;
+  source_col_ids.push_back(col_id);
+  std::string index_name1 = "Testing_Drop_Index_By_Name";
+  catalog->CreateIndex(TEST_DB_NAME, "department_table_01",
+                       source_col_ids, index_name1, false,
+                       IndexType::BWTREE, txn);
+  txn_manager.CommitTransaction(txn);
+
+  txn = txn_manager.BeginTransaction();
+  auto index_catalog = catalog::IndexCatalog::GetInstance()->GetIndexObject(
+              index_name1, txn);
+  // Check the effect of drop
+  // Most major check in this test case
+  EXPECT_EQ(catalog::IndexCatalog::GetInstance()->GetIndexObject(
+                index_name1, txn), index_catalog);
+
+  // Now dropping the index using the DropIndex functionality
+  catalog->DropIndex(index_name1,txn);
+  EXPECT_EQ(catalog::IndexCatalog::GetInstance()->GetIndexObject(
```

Comment:
Please check the same thing in another transaction as well.

Reply:
@poojanilangekar Do mean that I have to apply the same EXPECT_EQ in the transaction where I am dropping the table?
So that it can be checked before dropping the table that the index has really got dropped?

Thanks for the review. I am sorry that I couldn't do these in a single go.

Reply:
Yes.